packages/electron-updater/src/NsisUpdater.ts

```diff
@@ -235,6 +235,7 @@ async function _spawn(exe: string, args: Array<string>): Promise<any> {
   })
 }
 
-function hasQuotes(name: string): boolean {
-  return name.includes("'") || name.includes('"') || name.includes('`')
-}
\ No newline at end of file
+function isInvalid(name: string): boolean {
+  return name.includes("'") || name.includes('"') || name.includes('`') || name.includes('#') || name.includes('|') || name.includes('/') || name.includes('\') || name.includes(':') || name.includes('*') || name.includes('<') ||
+  name.includes('>') || name.includes('?')
+}
```

Comment:
Would checking for `name.includes(':')` not be problematic here?  I was under the impression that `destinationFile` and `packageFile` are an absolute path that includes the harddrive id, e.g.
```
C:\\Users\\User\\AppData\\Local\\myApp-updater
```
Matching `:` will cause the above path to be considered as invalid while it really should not be?

---
Also the same goes with `name.includes('\\')` where the path on windows will always contains `\\` 🤔 
```
C:\\Users\\User\\AppData\\Local\\myApp-updater
```

Reply:
@kevinlinv - I have reflected the requested changes!

Cheers! 🍰

Reply:
A comment on code quality, having an array of characters and using an array function would be much more readable and maintainable. 

e.g. 
```ts
...
const disallowedChars = [':','//', ...];
if(disallowedChars.some(char => name.includes(char)) return false;
return true;
...
```

Reply:
@b-zurg - I have made the requested changes! 🍰